spec/features/user_manages_space_spec.rb

```diff
@@ -0,0 +1,135 @@
+  end
+
+  it "user adds new space successfully" do
+    login_and_logout_with_warden do
+      visit root_path
+      click_link I18n.t("menu.new_space")
+
+      expect(page).to have_text("Legg til nytt lokale")
+
+      fill_in "space_title", with: "Space title!"
+      tom_select("select#space_space_type_id", option_id: space_type.id)
+
+      fill_in "space_address", with: "Ulefossvegen 32"
+      fill_in "space_post_number", with: "3730"
+      fill_in "space_post_address", with: "Skien"
+
+      tom_select("select#space_space_group_title", option_id: space_group.title)
+
+      click_button "Lag Lokale"
+      expect(page).to have_text("Space title!")
+    end
+  end
+
+  it "user edits space title" do
+    login_and_logout_with_warden do
+      visit space_path(id: space.id)
+      click_link "edit_basics"
+
+      fill_in "space_title", with: "New Space Title"
+
+      click_button "Lagre"
+      expect(page).to have_text("New Space Title")
+    end
+  end
+
+  it "user edits space basics unsuccessfully" do
+    login_and_logout_with_warden do
+      visit space_path(id: space.id)
+      click_link "edit_basics"
+
+      fill_in "space_title", with: ""
+      fill_in "space_url", with: "asd"
+      fill_in "space_address", with: ""
+      fill_in "space_post_number", with: ""
+      fill_in "space_post_address", with: ""
+
+      click_button "Lagre"
+      expect(page).to have_text("Title #{I18n.t('errors.messages.blank')}")
+      expect(page).to have_text("Nettside #{I18n.t('activerecord.errors.models.space.attributes.url.url')}")
+      expect(page).to have_text("Adresse #{I18n.t('errors.messages.blank')}")
+      expect(page).to have_text("Postnummer #{I18n.t('errors.messages.blank')}")
+      expect(page).to have_text("Poststed #{I18n.t('errors.messages.blank')}")
+    end
+  end
+
+  it "user adds new space contact" do
+    login_and_logout_with_warden do
+      visit space_path(id: space.id)
+      click_button "space_contacts_new"
+
+      fill_in "space_contact_title", with: "Space Contact Title"
+      fill_in "space_contact_telephone", with: "48484848"
+
+      click_button "Lagre"
+      expect(page).to have_text("Space Contact Title")
+    end
+  end
+
+  it "user adds new space contact unsuccessfully" do
+    login_and_logout_with_warden do
+      visit space_path(id: space.id)
+      click_button "space_contacts_new"
+
+      fill_in "space_contact_title", with: "Space Contact Title"
+
+      click_button "Lagre"
+      expect(page).to have_text("Telefon #{I18n.t('space_contact.at_least_one_error_message.telephone')}")
+      expect(page).to have_text("E-post #{I18n.t('space_contact.at_least_one_error_message.email')}")
+      expect(page).to have_text("Nettside #{I18n.t('space_contact.at_least_one_error_message.url')}")
+      expect(page).to have_text("Beskrivelse #{I18n.t('space_contact.at_least_one_error_message.description')}")
+    end
+  end
+
+  it "user adds new space contact with invalid website" do
```

Comment:
Is this one needed when we have the one above? I think if we know from the previous spec that 

1. You can add space contacts
2. You can get all the different error messages, if unsuccessful

Then testing for this as well seems a bit excessive - or something that should be done in a quicker model test instead?

What do you think?

Reply:
Yeah this was to essentially to test that `asd` wouldn't pass the URL validations. But it should be tested with a model spec